students/shibin_mathew/lesson4/trigram.py

```diff
@@ -1,34 +1,57 @@
+import random
+
 
 def read_file(filename):
-    # with open(path+'/'+filename, 'r+') as f:
-    #     data = f.read()
-    list_story = []
-    f = open(filename, 'r')
-    for line in f:
-            list_story.append(line.split())
-    return list_story
+    with open(filename, 'r+') as f:
+        data = f.read().replace('\n', ' ').replace('.', '').replace('--', ' ')
+        return data.split()
 
 
-def word_library(list_story, dict_word):
+def word_library(list_story):
     #  Adds new words to current list from the story
-    list_punctutation = [',', '.', ';', "'", '"']
-    print(list_story)
-    for i in range(0, len(list_story)):
-        if i + 1 <= len(list_story) and i + 2 <= len(list_story):
-            for word in dict_word:
-                # print(list_story[i] + ' ' + list_story[i+1])
-                if list_story[i] + ' ' + list_story[i + 1] == word:
-                    # if list_story[i + 2] not in list_punctutation:
-                        dict_word[word].append(list_story[i + 2])
-    return dict_word
+    list_punctutation = [',', '.', ';', "'", '"', '--']
+    my_word_dict = {}
+    for i in range(len(list_story)-2):
+        two_adj_words = list_story[i] + ' ' + list_story[i+1]
+        if list_story[i+2] not in list_punctutation:
+            third_word = list_story[i+2]
+            my_word_dict.setdefault(two_adj_words, []).append(third_word)
+    return my_word_dict
+
+
+def generate_story(sample_story, word_dict):
+    # Generate a story based on the trigram dictionary
+    rand_start = int(random.random()*len(word_dict))  # random starting point
+    story = ''
+    for i in range(len(sample_story)-2):
+        try:
+            if sample_story[i] + ' ' + sample_story[i+1] in word_dict.keys():
+                # print(sample_story[i] + ' ' + sample_story[i+1])
+                if i == 0:
+                    story += sample_story[i] + ' ' + sample_story[i+1] + ' ' + word_dict[sample_story[i] + ' ' +
```

Comment:
The assignment asks to begin with an arbitrary word pair, and then add a random following word from your `word_dict', but here (use random module), you're starting with the first three words of the input text